### ISSUE TITLE
'vim' not recognized

### ISSUE THREAD

**Issue by cliveong** [nus-cs2030/2021-s1#2]
Command prompt shows ''vim' is not recognized as an internal or external command,
operable program or batch file.' when i type vim -version

Also where do I put/how to use the .vimrc thing?

**Comment by yonggqiii** [comment]
Did you attempt to install Vim in WSL? Or did you only want to install it locally in Windows?

For the former, you should run Vim in WSL.

Otherwise, you might need to add vim to PATH.

If you're on WSL, you can just leave .vimrc in your home directory (~) and Vim will read the config directly.

**Comment by ClementNgZiQian** [comment]
I had this issue (cmd prompt showing 'vim not recognised as an internal or external command' just now too, but doing some googling, one of the ways suggested was to go to the directory containing vim.exe, which is most likely C:\Program Files (x86)\Vim\vim82, then run the install.exe as admin. It worked for me, hope it does for you.

**Comment by cliveong** [comment]
![image](https://user-images.githubusercontent.com/69676081/90243807-20ff8780-de62-11ea-93ed-d9288ec10907.png)
umm does this mean its installed in wsl?
![idekcorrectanot](https://user-images.githubusercontent.com/69676081/90244051-98351b80-de62-11ea-8161-9228f7c8d2f8.PNG)
does this count as leaving it in the home directory?

**Comment by rohit0718** [comment]
Hi there @cliveong! yup u have vim installed in your WSL. Note that the `-version` command doesnt work on WSL. You can find out more info about the version of `vim` installed using `vim --version`. As for the home directory, once you have run your `wsl` command on command prompt, try running `echo $HOME`. It should print the WSL home directory, which is where you would need to place your `.vimrc`. The home directory you shoudl place your `.vimrc` file is the WSL's one, and not the windows one!

**Comment by cliveong** [comment]
![image](https://user-images.githubusercontent.com/69676081/90257714-18b34680-de7a-11ea-8faf-eabfc3d11686.png)
Is this correct?
![image](https://user-images.githubusercontent.com/69676081/90257786-31bbf780-de7a-11ea-971e-20c6720bfcb8.png)
This is what I got

**Comment by rohit0718** [comment]
sry i realised the `echo $HOME` doesnt help here! yup the folder u have open in the file explorer looks correct!

**Comment by yonggqiii** [comment]
@rohit0718 you're really fast 👍🏻👍🏻

@cliveong word of caution, avoid accessing your files in WSL via the Windows File Explorer. Instead, use the command line and the commands taught in the videos :D

**Comment by cliveong** [comment]
Idk why but I got permission denied when typing in 'vim ~/.vimrc', even when running as admin, it instead copied the name and made an empty file. I copied and pasted the text inside to the .vimrc file and it seems  to work fine now

**Comment by yonggqiii** [comment]
Did you drag .vimrc into WSL from windows? Ubuntu doesn't like that.

Instead, you should navigate to your windows desktop via WSL and enter cd /mnt/c blablabla. That way, you can copy your .vimrc into ~

Alternatively, you could try sudo chmod 777 ~/.vimrc though I wouldn't recommend it

**Comment by cliveong** [comment]
umm not really i tried the command which produced an empty .vimrc so I copy pasted the stuff into the empty .vimrc file
![image](https://user-images.githubusercontent.com/69676081/90270077-34274d00-de8c-11ea-9a1b-fcd46dfc73dd.png)
How do I check if it is working?

**Comment by yonggqiii** [comment]
@cliveong Ah I see. Try opening Vim and type some random stuff, if you see line numbers on the left, you're more or less good to go :D

**Comment by cliveong** [comment]
So I guess this one is wrong?
![image](https://user-images.githubusercontent.com/69676081/90270450-cc253680-de8c-11ea-9db5-42881b9fd7e2.png)

**Comment by rohit0718** [comment]
What is the output for `ls -l .vimrc`? (type it in the home directory)

**Comment by cliveong** [comment]
![image](https://user-images.githubusercontent.com/69676081/90271809-f972e400-de8e-11ea-8a07-5f28e8373a87.png)

**Comment by rohit0718** [comment]
oh i think the problem is that youre editing the file on windows! when accessing files on WSL you should try to do it directly on WSL itself! try running `chmod +r .vimrc` in the home directory and running `vim` again? (chmod is a command for changing permissions and here you are adding the read (r) permissions)

**Comment by cliveong** [comment]
![image](https://user-images.githubusercontent.com/69676081/90272476-10660600-de90-11ea-84d2-714dca33d355.png)
Is this correct?

**Comment by rohit0718** [comment]
hahah close! ok so now its reading the `.vimrc` but cause u edited with windows editor, the line endings are different (CRLF in windows vs LF in linux - not sure why its diff lol). I think best thing to do here would jus be to remove the file and paste in the contents through WSL itself! 

`rm .vimrc` to remove the file. `vim .vimrc` to create and enter the file. copy the contents of the vimrc file from github, press `i` in the editor to enter `insert` mode, then paste in the contents with `ctrl+shift+v`. once done hit the `esc` key, and type in `:wq` and hit enter. (`:wq` saves the file and quits)

**Comment by cliveong** [comment]
Wow i think i got it
![image](https://user-images.githubusercontent.com/69676081/90273441-90d93680-de91-11ea-86fe-5c95c87b13b2.png)
Pls tell me this is correct lol

**Comment by rohit0718** [comment]
yup thats right!

**Comment by cliveong** [comment]
Nice! Thanks a lot @rohit0718

**Comment by cmukid** [comment]
Maybe we shouldn't ask students to install Vim in Windows. Just use the one in  WSL which is already installed. Having 2 Vims cause confusion. 

IMHO, a GUI text editor is pointless. Either use a full-fledged IDE or else stick to a non-GUI text-editor.

**Comment by yonggqiii** [comment]
@Terence-Sim I agree! With WSL I can't see a reason to also have Vim installed in Windows.

That aside, Visual Studio Code users are gonna feel pain 😩😩😩😩
